fitbenchmarking/fitbenchmarking
# 216 add dfo gn

#### Description of Work

Resolves #216 

#### Testing Instructions

1. Add `dfogn` to the software list in an example script
2. Check output is as expected (especially on `simple_tests`)

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?
